lib/ff_dpdk_if.c

```diff
@@ -1416,7 +1435,5 @@ ff_rss_check(void *softc, uint32_t saddr, uint32_t daddr,
     uint32_t hash = toeplitz_hash(sizeof(default_rsskey_40bytes),
         default_rsskey_40bytes, datalen, data);
 
-    return ((hash & (reta_size - 1)) % qconf->nb_procs) == qconf->proc_id;
+    return ((hash & (reta_size - 1)) % qconf->nb_procs) == queueid;
```

Comment:
This is wrong, it shouldn't perform a modulus on `qconf->nb_procs`,  it should be the port's number of queues.